### DISCUSSION TITLE
Is it possible to use partitions for observable source assets

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18389] [Q&A] [answered]
Is it possible to use partitions for observable source assets? <https://docs.dagster.io/concepts/assets/asset-observations#observable-source-assets>

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1701288724881429?thread_ts=1701288724.881429&cid=C01U954MEER).*

**Comment by smackesey** [comment, marked as answer]
Yes it is-- the `@observable_source_asset` decorator takes a `partitions_def` argument. If it's partitioned, then you should return a `DataVersionsByPartition` object in the observe function.

```python
from dagster import DataVersionsByPartition, StaticPartitionsDefinition, observable_source_asset

@observable_source_asset(
    partitions_def=StaticPartitionsDefinition(["a", "b"])
)
def foo():
    return DataVersionsByPartition({"a": "alpha", "b": "beta"})
```

It does look like we have a documentation hole here, so thanks for the report.
